library/src/main/java/com/schibsted/spain/barista/cleardata/ClearDatabaseRule.java

```diff
@@ -0,0 +1,112 @@
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+/**
+ * This rule clears all app's Databases before running each test
+ */
+public class ClearDatabaseRule implements TestRule {
+
+  private static final String[] UNINTERESTING_FILENAME_SUFFIXES = new String[] {
+      "-journal",
+      "-shm",
+      "-uid",
+      "-wal"
+  };
+
+  @Override
+  public Statement apply(final Statement base, Description description) {
+    return new Statement() {
+      @Override
+      public void evaluate() throws Throwable {
+        clearData();
+        base.evaluate();
+        clearData();
+      }
+    };
+  }
+
+  private void clearData() {
+    for (File dbFile : tidyDatabaseList(getDatabaseFiles())) {
+      SQLiteDatabase database = performOpen(dbFile, 0);
+      for (String table : getTableNames(database)) {
+        database.delete(table, null, null);
+      }
+      database.close();
+    }
+  }
+
+  private List<File> getDatabaseFiles() {
+    Context context = InstrumentationRegistry.getTargetContext();
+    List<File> databaseFiles = new ArrayList<>();
+    for (String databaseName : context.databaseList()) {
+      databaseFiles.add(context.getDatabasePath(databaseName));
+    }
+    return databaseFiles;
+  }
+
+  private List<String> getTableNames(SQLiteDatabase database) throws SQLiteException {
+    Cursor cursor = database.rawQuery("SELECT name FROM sqlite_master WHERE type IN (?, ?)",
+        new String[] { "table", "view" });
+    try {
+      List<String> tableNames = new ArrayList<>();
+      while (cursor.moveToNext()) {
+        tableNames.add(cursor.getString(0));
+      }
+      return tableNames;
+    } finally {
+      cursor.close();
+    }
+  }
+
+  private SQLiteDatabase performOpen(File databaseFile, int options) {
+    int flags = SQLiteDatabase.OPEN_READWRITE;
+
+    return SQLiteDatabase.openDatabase(
+        databaseFile.getAbsolutePath(),
+        null /* cursorFactory */,
+        flags);
+  }
+
+  /**
+   * Attempt to smartly eliminate uninteresting shadow databases such as -journal and -uid.  Note
+   * that this only removes the database if it is true that it shadows another database lacking
+   * the uninteresting suffix.
+   *
+   * @param databaseFiles Raw list of database files.
+   * @return Tidied list with shadow databases removed.
+   */
+  private static List<File> tidyDatabaseList(List<File> databaseFiles) {
+    Set<File> originalAsSet = new HashSet<>(databaseFiles);
+    List<File> tidiedList = new ArrayList<>();
+    for (File databaseFile : databaseFiles) {
+      String databaseFilename = databaseFile.getPath();
+      String sansSuffix = removeSuffix(databaseFilename, UNINTERESTING_FILENAME_SUFFIXES);
+      if (sansSuffix.equals(databaseFilename) || !originalAsSet.contains(new File(sansSuffix))) {
+        tidiedList.add(databaseFile);
+      }
+    }
+    return tidiedList;
+  }
+
+  private static String removeSuffix(String str, String[] suffixesToRemove) {
```

Comment:
What do you think about calling this method `filterUninterestingSuffixes`? And I suggest to avoid passing the list of suffixest to remove. It seems that they will be always the same:  `UNINTERESTING_FILENAME_SUFFIXES`.